authnz/src/main/java/io/stargate/auth/AuthorizationService.java

```diff
@@ -42,7 +44,8 @@ ResultSet authorizedDataRead(
       String keyspace,
       String table,
       List<TypedKeyValue> typedKeyValues,
-      SourceAPI sourceAPI)
+      SourceAPI sourceAPI,
+      Map<String, String> headers)
```

Comment:
I'm not sure the `headers` is justified in this context. I believe all the information required for authorization should be contained in the `AuthenticationSubject` object, which is already defined in this method.

Reply:
Basically, I think we ought to add tenant ID (plus whatever else is needed) to `AuthenticationSubject`. We already reference tenant ID in `RequestToHeadersMapper` so it's not completely transparent and it would be nice to make tenant ID handling explicit.

Reply:
If the list of extra properties is not know, we could add a `properties` object of type `Map<String, String>` to `AuthenticationSubject`, but I think it is important to keep authorization inputs together. That is the pattern in the JRE auth APIs, cf. `javax.security.auth.Subject`. WDYT?

Reply:
Please note that `headers` are just passing through. They are not used in the `Stargate` implementations at all. Also, the `tenant_id` is needed not only for IAM but also for example billing. By including that information in the `AuthenticationSubject` we will hide it from other callers. I think that this is one of the main intentions of why we want to pass it separately.

Reply:
I agree with @dimas-b that a more generic term would be better but perhaps we should create an issue in order to come back and refactor?